pixel-avatar-hardhat/contracts/pxl-avatars.sol

```diff
@@ -0,0 +1,1803 @@
+    function sub(
+        uint256 a,
+        uint256 b,
+        string memory errorMessage
+    ) internal pure returns (uint256) {
+        unchecked {
+            require(b <= a, errorMessage);
+            return a - b;
+        }
+    }
+
+    /**
+     * @dev Returns the integer division of two unsigned integers, reverting with custom message on
+     * division by zero. The result is rounded towards zero.
+     *
+     * Counterpart to Solidity's `/` operator. Note: this function uses a
+     * `revert` opcode (which leaves remaining gas untouched) while Solidity
+     * uses an invalid opcode to revert (consuming all remaining gas).
+     *
+     * Requirements:
+     *
+     * - The divisor cannot be zero.
+     */
+    function div(
+        uint256 a,
+        uint256 b,
+        string memory errorMessage
+    ) internal pure returns (uint256) {
+        unchecked {
+            require(b > 0, errorMessage);
+            return a / b;
+        }
+    }
+
+    /**
+     * @dev Returns the remainder of dividing two unsigned integers. (unsigned integer modulo),
+     * reverting with custom message when dividing by zero.
+     *
+     * CAUTION: This function is deprecated because it requires allocating memory for the error
+     * message unnecessarily. For custom revert reasons use {tryMod}.
+     *
+     * Counterpart to Solidity's `%` operator. This function uses a `revert`
+     * opcode (which leaves remaining gas untouched) while Solidity uses an
+     * invalid opcode to revert (consuming all remaining gas).
+     *
+     * Requirements:
+     *
+     * - The divisor cannot be zero.
+     */
+    function mod(
+        uint256 a,
+        uint256 b,
+        string memory errorMessage
+    ) internal pure returns (uint256) {
+        unchecked {
+            require(b > 0, errorMessage);
+            return a % b;
+        }
+    }
+}
+
+// ---
+
+pragma solidity ^0.8.7;
+
+interface LootInterface {
+    function ownerOf(uint256 tokenId) external view returns (address owner);
+}
+
+contract PixelAvatars is ERC721Enumerable, ReentrancyGuard, Ownable {
+    using SafeMath for uint256;
+
+    // TODO: set this to the ipfs base uri before launch
+    string public baseURI =
+        "https://g69d6ix8sb.execute-api.us-east-1.amazonaws.com/dev/api/tokens/";
+
+    //Original Developer Dao Contract
+    address public DevDaoAddress = 0x25ed58c027921E14D86380eA2646E3a1B5C55A8b; // TODO: Change this to mainnet Developer Dao Contract
+    LootInterface DevDaoContract = LootInterface(DevDaoAddress);
+
+    constructor() ERC721("Pixel Avatars", "PXLAVTR") {}
+
+    function _baseURI() internal view virtual override returns (string memory) {
+        return baseURI;
+    }
+
+    function setBaseURI(string memory newBaseURI) public onlyOwner {
+        baseURI = newBaseURI;
+    }
+
+  
+
+    //Private sale minting (reserved for DevDao owners)
+    function mintWithDevDaoToken(uint256 DevDaoTokenId)
+        public
+        payable
```

Comment:
Why is this payable tho?

Reply:
good catch. Not sure why this was kept payable on my reference contract. I'll remove this.

Reply:
Reposting my comment as etherinsight again 😅 

good catch. Not sure why this was kept payable on my reference contract. I'll remove this.